docs/source/protocols/api/mqtt/mqtt_switch.rst

```diff
@@ -0,0 +1,120 @@
+.. |yes| image:: ../../images/yes.png
+.. |no| image:: ../../images/no.png
+
+.. role:: underline
+   :class: underline
+
+Mqtt_switch
+===========
+
+Behaves like a normal switch within pilight, but instead of sending and receiving via rf, it publishes an mqtt message when its state changes and changes its state if an ``on`` or ``off`` message is received via mqtt.
+
++------------------+-------------+
+| **Feature**      | **Support** |
++------------------+-------------+
+| Sending          | |no|        |
++------------------+-------------+
+| Receiving        | |no|        |
++------------------+-------------+
+| Config           | |yes|       |
++------------------+-------------+
+
+.. rubric:: Supported Brands
+
+*None*
+
+.. rubric:: Sender Arguments
+
+.. code-block:: console
+
+   -t --on                  send an on signal
+   -f --off                 send an off signal
+   -i --id=id               control a device with this id
+
+.. rubric:: Config
+
+.. code-block:: json
+   :linenos:
+
+   {
+     "devices": {
+       "mqtt1": {
+         "protocol": [ "mqtt_switch" ],
+         "id": [{
+           "id": 1
```

Comment:
Are these the `clientid` as used in the hardware module?

Reply:
No, this is just the id of the mqtt_switch device in the config..

Reply:
To be a bit more accurate, the device's id is used when constructing a unique clientid, by combining the protocol name with the device id and the role (pub or sub).